clientv3/integration/cluster_test.go

```diff
@@ -127,3 +127,36 @@ func TestMemberUpdate(t *testing.T) {
 		t.Errorf("urls = %v, want %v", urls, resp.Members[0].PeerURLs)
 	}
 }
+
+func TestClientSync(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	cli := clus.Client(0)
+	if err := cli.Sync(); err != nil { // initialize
+		t.Fatal(err)
+	}
+
+	clus.AddMember(t)
+	clus.AddMember(t)
+	clus.WaitLeader(t)
+
+	// sync to grab the new members
+	if err := cli.Sync(); err != nil {
+		t.Fatal(err)
+	}
+
+	// stop the original member
+	clus.Members[0].Stop(t)
+	clus.WaitLeader(t)
+
+	// rpc will only go through if the sync worked
```

Comment:
This will fail because adding more endpoints doesn't update the unavailable pinned endpoint. Do we expect users to detect the unavailable endpoint and change the pinned one manually?